crates/fuzzing/src/oracles.rs

```diff
@@ -479,3 +476,189 @@ pub fn table_ops(config: crate::generators::Config, ops: crate::generators::tabl
         }
     }
 }
+
+/// Do we accept the given wasm module for differential fuzzing between Wasmtime and wasmi? We want
+/// to return early for modules that don't meet our criteria so that the fuzzer learns that these
+/// are not interesting (as they will not lead to additional coverage).
+///
+/// If accepted, returns the name of the memory we will diff and the function we will call.
+fn differential_wasmi_accept_module(wasm: &[u8]) -> Option<(String, String)> {
+    // Find the names of the first memory and first export.
+    let mut memory_name = None;
+    let mut func_name = None;
+    let parser = wasmparser::Parser::new(0);
+    for payload in parser.parse_all(&wasm[..]) {
+        let payload = match payload {
+            Ok(p) => p,
+            Err(_) => return None,
+        };
+        match payload {
+            wasmparser::Payload::ExportSection(mut e) => {
+                for _ in 0..e.get_count() {
+                    let exp = match e.read() {
+                        Ok(exp) => exp,
+                        Err(_) => return None,
+                    };
+                    match exp.kind {
+                        wasmparser::ExternalKind::Memory => {
+                            memory_name = Some(exp.field.to_string());
+                        }
+                        wasmparser::ExternalKind::Function => {
+                            func_name = Some(exp.field.to_string());
+                        }
+                        _ => {}
+                    }
+                    if memory_name.is_some() && func_name.is_some() {
+                        break;
+                    }
+                }
+            }
+            wasmparser::Payload::MemorySection(mut m) => {
+                for _ in 0..m.get_count() {
+                    let mem = match m.read() {
+                        Ok(mem) => mem,
+                        Err(_) => return None,
+                    };
+                    match mem {
+                        wasmparser::MemoryType::M32 { limits, .. } => {
+                            if limits.initial > 100 {
+                                return None;
+                            }
+                            if limits.maximum.is_none() || limits.maximum.unwrap() > 100 {
```

Comment:
Does wasmi not support larger memories? Or is this to just limit how much memory we're comparing?

Reply:
Indeed, this is both to limit the comparison time and to reduce memory usage; I was getting OOMs otherwise during fuzzing.